solver/src/http_solver.rs

```diff
@@ -190,6 +203,16 @@ impl HttpSolver {
             .get_token_infos(addresses.as_slice())
             .await;
 
+        let price_estimates: HashMap<H160, Result<BigRational>> = addresses
+            .iter()
+            .cloned()
+            .zip(
+                self.price_estimator
```

Comment:
nit: estimate_price and the other async operation (get_token_infos) could happen in parallel and be awaited on via a join!

Reply:
Wouldn't I need to wrap these two futures into some common abstraction in order to aggregate them?

Reply:
In this case no.

```rust
let (token_infos, price_estimates) = futures::join!(
    self.token_info_fetcher
        .get_token_infos(addresses.as_slice()),
    self.price_estimator
        .estimate_prices(addresses.as_slice(), addresses[0]),
);

let price_estimates: HashMap<H160, Result<BigRational>> =
    addresses.iter().cloned().zip(price_estimates).collect();
```
I am using the join! macro https://docs.rs/futures/0.3.13/futures/future/fn.join.html . There is also a plain function https://docs.rs/futures/0.3.13/futures/macro.join.html . The macro has the advantage of being able to take any number of arguments.

Reply:
Thanks, that seems to work (even if my editor complains that join! is a unresolved macro call).

Reply:
I think you have to enable the rust analyzer setting "proc macro".

Reply:
Thanks, it seems it is on:

![Screenshot_20210330_115124](https://user-images.githubusercontent.com/624308/112977690-41a8f300-914e-11eb-857f-b05d0c65075f.png)

But this is minor. Thanks!